packages/test/test-end-to-end-tests/src/test/refreshSerializedStateManager.spec.ts

```diff
@@ -133,6 +133,7 @@ describeCompat("Snapshot refresh at loading", "NoCompat", (getTestObjectProvider
 		await timeoutAwait(getLatestSnapshotInfoP.promise, {
 			errorMsg: "Timeout on waiting for getLatestSnapshotInfo",
 		});
+		await provider.ensureSynchronized();
```

Comment:
This will ensure that pending ops are applied and not being carried to the next stashed state. There is a scenario I'm hitting that I don't know exactly how to handle which is exactly this test without ensureSynchronized().
I'm adding a condition after fetching the new snapshot to check if the container is dirty. If it's not dirty we should be able to update this.snapshot so in theory it prevents using a newer snapshot if pending ops haven't being applied.
This was not the case, since it seems the snapshot fetching finishes before we even instantiate the runtime and does not give time to switch the container state to dirty.